dataset/src/main/scala/frameless/TypedColumn.scala

```diff
@@ -602,6 +616,29 @@ abstract class AbstractTypedColumn[T, U]
 }
 
 
+sealed class SortedTypedColumn[T, U](val expr: Expression)(
+  implicit
+  val uencoder: TypedEncoder[U]
+) extends UntypedExpression[T] {
+
+  def this(column: Column)(implicit e: TypedEncoder[U]) {
+    this(FramelessInternals.expr(column))
+  }
+
+  def untyped: Column = new Column(expr)
+}
+
+object SortedTypedColumn {
+  implicit def defaultAscending[T, U : CatalystOrdered](typedColumn: TypedColumn[T, U]): SortedTypedColumn[T, U] =
+      new SortedTypedColumn[T, U](new Column(SortOrder(typedColumn.expr, Ascending)))(typedColumn.uencoder)
```

Comment:
If this works and looks simpler:
` new SortedTypedColumn[T, U](typedColumn.untyped.asc)(typedColumn.uencoder)`

Reply:
^

Reply:
^

Reply:
how frustrating. sorry. fixed.